apps/dashboard/app/batch-payments/server-actions.ts

```diff
@@ -159,3 +181,82 @@ export const processPaymentsServerAction = async (
     responsePayload: null,
   }
 }
+
+export const validatePaymentDetail = async (
+  TotalAmount: TotalAmountForWallets,
+): Promise<{
+  error: boolean
+  message: string
+  responsePayload: {
+    transactionAmount: {
+      totalAmountForBTCWallet: number
+      totalAmountForUSDWallet: number
+    }
+    userWalletBalance: {
+      btcWalletBalance: number
+      usdWalletBalance: number
+    }
+  } | null
+}> => {
+  // TODO move this logic to client side by setting up apollo clint
```

Comment:
what would be the benefits of client side logic?

Reply:
Not completely sure about moving it to the client side, but the main reason would be to avoid unnecessary API calls, since this server action only checks if the user's balance exceeds the payable amount (doesn't require much compute). We already have the user balance and payable amount available on the client side. The only reason for processing this on the server side currently is the realTime price query. However, we can manage this on the client side by setting up an Apollo provider on client and directly calling the realTimePriceQuery or price subscription there, instead of making a call to the server and then to the Price APIs.
thinking on this not completely sure...